
From: Adrian Bunk <bunk@fs.tum.de>

Rationale:

- if gcc 3.4 can't inline a function marked as "inline" that's a strong
  hint that further investigation is required

- I strongly prefer a compile error over a potential runtime problem

Signed-off-by: Adrian Bunk <bunk@fs.tum.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/compiler-gcc3.h |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN include/linux/compiler-gcc3.h~define-inline-as-__attribute__always_inline-also-for-gcc-=-34 include/linux/compiler-gcc3.h
--- 25/include/linux/compiler-gcc3.h~define-inline-as-__attribute__always_inline-also-for-gcc-=-34	2004-07-11 21:53:19.009502216 -0700
+++ 25-akpm/include/linux/compiler-gcc3.h	2004-07-11 21:53:19.013501608 -0700
@@ -3,7 +3,7 @@
 /* These definitions are for GCC v3.x.  */
 #include <linux/compiler-gcc.h>
 
-#if __GNUC_MINOR__ >= 1  && __GNUC_MINOR__ < 4
+#if __GNUC_MINOR__ >= 1
 # define inline		__inline__ __attribute__((always_inline))
 # define __inline__	__inline__ __attribute__((always_inline))
 # define __inline	__inline__ __attribute__((always_inline))
_
